internal/app/admin/http/handler.go

```diff
@@ -109,52 +109,73 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
+	resp := marshallableCache{}
+	for _, key := range keys {
+		resource, err := cache.FetchReadOnly(key)
+		if err != nil {
+			w.WriteHeader(http.StatusNotFound)
+			fmt.Fprintf(w, "no resource for key %s found in cache.\n", key)
+			continue
+		}
+		resp.Cache = append(resp.Cache, resourceToPayload(key, resource)...)
+	}
+	resourceString, err := stringify.InterfaceToString(resp)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "unable to convert resource to string.\n")
+		return
+	}
+
+	if len(resp.Cache) > 0 {
+		fmt.Fprintf(w, "%s\n", resourceString)
+	}
+
+}
+
 func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		c := orchestrator.Orchestrator.GetReadOnlyCache(*o)
-		resp := marshallableCache{}
-		// If no key is provided, output the entire cache.
-		if cacheKey == "" {
+		var keysToPrint []string
+
+		// If wildcard suffix provided, output all cache entries that match given prefix. If no key is
+		// provided, output the entire cache.
+		containsWildcardSuffix := strings.HasSuffix(cacheKey, "*")
+		if cacheKey == "" || containsWildcardSuffix {
+			rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
+
+			// Retrieve all keys
 			keys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
 			if err != nil {
 				w.WriteHeader(http.StatusInternalServerError)
 				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
 				return
 			}
 
-			for key := range keys {
-				resource, err := c.FetchReadOnly(key)
-				if err != nil {
-					continue
+			// Find keys that match prefix of wildcard
+			if containsWildcardSuffix {
+				for potentialMatchKey := range keys {
+					if !strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
+						delete(keys, potentialMatchKey)
+					}
+				}
+				if len(keys) == 0 {
+					w.WriteHeader(http.StatusNotFound)
+					fmt.Fprintf(w, "no resource for key %s found in cache.\n", cacheKey)
+					return
 				}
-				resp.Cache = append(resp.Cache, resourceToPayload(key, resource)...)
-			}
-			resourceString, err := stringify.InterfaceToString(resp)
-			if err != nil {
-				return
 			}
-			fmt.Fprintf(w, "%s\n", resourceString)
-			return
-		}
 
-		// Otherwise return the cache entry corresponding to the given key.
-		resource, err := c.FetchReadOnly(cacheKey)
-		if err != nil {
-			w.WriteHeader(http.StatusNotFound)
-			fmt.Fprintf(w, "no resource for key %s found in cache.\n", cacheKey)
-			return
-		}
-
-		resp.Cache = append(resp.Cache, resourceToPayload(cacheKey, resource)...)
-		resourceString, err := stringify.InterfaceToString(resp)
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			fmt.Fprintf(w, "unable to convert resource to string.\n")
-			return
+			// Add matched keys to print slice
+			for key := range keys {
+				keysToPrint = append(keysToPrint, key)
```

Comment:
instead of deleting keys earlier, we can instead add keys to `keysToPrint` directly.